google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSource.scala

```diff
@@ -31,13 +31,27 @@ private final class GooglePubSubSource(projectId: String,
       def fetch(implicit mat: Materializer): Unit = {
         import mat.executionContext
 
-        val req = session.getToken().flatMap { token =>
-          httpApi.pull(project = projectId, subscription = subscription, accessToken = token, apiKey = apiKey)
-        }
-        req.onComplete { tr =>
-          callback.invoke(tr -> mat)
+        if (httpApi.isEmulated) {
+          httpApi
+            .pull(project = projectId, subscription = subscription, maybeAccessToken = None, apiKey = apiKey)
+            .onComplete { tr =>
```

Comment:
To make it obvious that the `onComplete` is shared between both branches you might go back to the `val req` by assigning the if clause to it.

Reply:
Sure, changed. Is this what you meant?